### PR TITLE
feat: create YARN znode with secure ACLs

### LINKED ISSUE #335: ZooKeeper: YARN znodes should be protected
The YARN znodes are not protected by any ACLs and can be modified by any user with a valid Kerberos ticket:
- `/yarn-leader-election`
- `/rmstore` (used by YARN RM)

Based on Hortonworks [ZooKeeper ACLs Best Practices: YARN](https://docs.cloudera.com/HDPDocuments/HDP3/HDP-3.1.0/zookeeper-acls/content/zookeeper_acls_best_practices_yarn.html) the ACLs should be set as follow:
- `/yarn-leader-election`: `world:anyone:r`, `sasl:rm:rwcda`
- `/rmstore`: `world:anyone:r`, `sasl:rm:rwcda`

### PR DESCRIPTION
**Pull request by leopaul36**
#### Which issue(s) this PR fixes

Fixes #335 

#### Agreements

- [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Comment by gboutry** [comment]
Aren't there any configuration option available to setup znode ACLs ?

**Comment by leopaul36** [comment]
> Aren't there any configuration option available to setup znode ACLs ?

I couldn't find any. I only applied the recommendations found at https://docs.cloudera.com/HDPDocuments/HDP3/HDP-3.1.0/zookeeper-acls/content/zookeeper_acls_best_practices_yarn.html

File: tdp_vars_defaults/yarn/yarn.yml [hunk 1]
```diff
@@ -45,6 +45,8 @@ yarn_site:
   yarn.resourcemanager.system-metrics-publisher.enabled: "true"
   yarn.resourcemanager.recovery.enabled: "true"
   yarn.resourcemanager.store.class: "org.apache.hadoop.yarn.server.resourcemanager.recovery.ZKRMStateStore"
+  yarn.resourcemanager.zk-state-store.parent-path: "/rmstore"
+  yarn.resourcemanager.ha.automatic-failover.zk-base-path: "/yarn-leader-election"
```
Comment (gboutry) [inline tdp_vars_defaults/yarn/yarn.yml:49]:
```suggestion
  yarn.resourcemanager.ha.automatic-failover.zk-base-path: "/yarn-leader-election"
  hadoop.zk.acl: world:anyone:r,sasl:rm:rwcda
```

I tested a deployment while adding this property, it creates the zNodes with the proper ACLs (I had to get the jaas part from this PR for this to work)

```
[zk: master-01.tdp:2181,master-02.tdp:2181,master-03.tdp:2181(CONNECTED) 0] ls /
[hiveserver2, zookeeper, yarn-leader-election, hadoop-ha, rmstore, hadoop-kms, hbase]
[zk: master-01.tdp:2181,master-02.tdp:2181,master-03.tdp:2181(CONNECTED) 1] getAcl /yarn-leader-election
'world,'anyone
: r
'sasl,'rm
: cdrwa
[zk: master-01.tdp:2181,master-02.tdp:2181,master-03.tdp:2181(CONNECTED) 2] getAcl /rmstore             
'world,'anyone
: r
'sasl,'rm
: cdrwa
```

Comment (leopaul36) [reply tdp_vars_defaults/yarn/yarn.yml:49]: Good catch, I did not think to look in the core-site properties.

I'll refactor the PR only adding this property and the jaas for the YARN RM

Comment (leopaul36) [reply tdp_vars_defaults/yarn/yarn.yml:50]: Done in the latest push force

**Review by gboutry — APPROVED** [review]

**Review by Nuttymoon — APPROVED** [review]

**Pull request merged**